### PR TITLE
house detail service added

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR I have implemented the service to get the house detail data from the API and then show this data into the "HouseDetailFragment". Also I implement the test for this feature, and make a refactor to the tests of the other features, mainly take out the "ObsoleteCoroutinesApi" and testing some new cases.
Gif showing the app (Nexus 5X):
houseDetailServicePRNexus5X

Gif showing the app (Nexus S):
houseDetailServicePRNexusS

Tests image:
image
Coverage image:
image

### CODE REVIEW

File: app/src/main/java/com/globant/harrypotterapp/fragment/HouseDetailFragment.kt [hunk 1]
```diff
@@ -9,49 +9,59 @@ import android.widget.Toast
...
+            binding.houseDetailFragmentHouseGhost.text = getString(R.string.house_detail_fragment_ghost_text, houseGhost)
+            binding.houseDetailFragmentFounder.text = getString(R.string.house_detail_fragment_founder_text, founder)
+            when (name) {
                 GRYFFINDOR -> binding.houseDetailFragmentImage.setImageResource(R.drawable.gryffindor)
                 RAVENCLAW -> binding.houseDetailFragmentImage.setImageResource(R.drawable.ravenclaw)
                 SLYTHERIN -> binding.houseDetailFragmentImage.setImageResource(R.drawable.slytherin)
                 HUFFLEPUFF -> binding.houseDetailFragmentImage.setImageResource(R.drawable.hufflepuff)
-                else -> {
-                    binding.houseDetailFragmentImage.setImageResource(R.drawable.hogwarts)
-                    showErrorMessage()
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/harrypotterapp/fragment/HouseDetailFragment.kt:44]: shouldnt you call `showErrorMessage` on else branch?

Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/fragment/HouseDetailFragment.kt:44]: Yes, done!

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -7,6 +7,12 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="AppTheme.NoActionBar">
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml]: be careful here, because there already exists a native NoActionBar theme

**Review by SartiMau — CHANGES_REQUESTED** [review]
Put the data and status en each ViewModel, do not use a general one

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (benjaglobant) [reply app/src/main/res/values/styles.xml]: Yes! Refactored!

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**